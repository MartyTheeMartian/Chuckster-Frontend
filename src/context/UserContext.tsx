import { createContext, useState, useMemo } from 'react';


interface UserData {
  authenticated: boolean;
  email: string;
  password: string;
  jwtToken: string;
}

interface UserContextInterface {
  user: UserData;
  setUser: React.Dispatch<React.SetStateAction<UserData>>;
}

interface ChildrenProps {
  children: JSX.Element | JSX.Element[];
}

const defaultUser = {
  authenticated: false,
  email: '',
  password: '',
  jwtToken: '',
};

const defaultUserContext: UserContextInterface = {
  user: defaultUser,
  setUser: () => {},
};


export const UserContext = createContext(defaultUserContext);

export default function UserContextProvider({ children }: ChildrenProps): JSX.Element {

  const [user, setUser] = useState(defaultUser);

  const userContextValue = useMemo(
    () => ({
      user,
      setUser,
    }),
    [user]
  );

  return (
    <UserContext.Provider value={userContextValue}>
      { children }
    </UserContext.Provider>
  );
}
