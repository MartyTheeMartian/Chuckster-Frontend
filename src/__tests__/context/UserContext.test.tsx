import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserContext from 'context/UserContext';
import { BrowserRouter as Router } from 'react-router-dom';


it('renders App component', () => {
  const result = shallow(
    <UserContext>
      <Router />
    </UserContext>
  );
  expect(toJson(result)).toMatchSnapshot();
});
