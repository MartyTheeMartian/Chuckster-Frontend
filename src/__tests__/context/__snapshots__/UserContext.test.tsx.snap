// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component 1`] = `
<ContextProvider
  value={
    {
      "setUser": [Function],
      "user": {
        "authenticated": false,
        "email": "",
        "jwtToken": "",
        "password": "",
      },
    }
  }
>
  <BrowserRouter />
</ContextProvider>
`;
